### PR TITLE
fix: render bar chart only after fetched

### PR DESCRIPTION
**Pull request by pdp2121**
## High Level Overview of Change

Only show the bar chart after data is fully fetched. Otherwise, a loading icon will be shown.

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

### TypeScript/Hooks Update

- [ ] Updated files to React Hooks
- [ ] Updated files to TypeScript

### Before
When data is not fully fetched, bar chart looks like this:
Screen Shot 2022-09-21 at 3 52 09 PM

### After
A loading icon will be shown when data is not fully fetched:
Screen Shot 2022-09-22 at 1 50 23 PM (2)

### CODE REVIEW

**Review by jonathanlei — APPROVED** [review]

File: src/containers/Network/UpgradeStatus.tsx [hunk 1]
```diff
@@ -136,7 +138,7 @@ export const UpgradeStatus = () => {
       <div className="wrap">
         <NetworkTabs selected="upgrade_status" />
         <div className="upgrade_status">
-          {vList && (
+          {validatorCount && colorLoaded && (
```
Comment (ckniffen) [inline src/containers/Network/UpgradeStatus.tsx:141]: You could check if there is a current version instead of adding a new prop

**Review by ckniffen — APPROVED** [review]

File: src/containers/Network/UpgradeStatus.tsx [hunk 2]
```diff
@@ -111,32 +111,28 @@ export const UpgradeStatus = () => {
...
+            )}
+          </span>
+        </>
       </div>
-
       <div className="wrap">
         <NetworkTabs selected="upgrade_status" />
         <div className="upgrade_status">
-          {vList && (
+          {validatorCount && stableVersion && (
```
Comment (mvadari) [inline src/containers/Network/UpgradeStatus.tsx:135]: There should be a "Loading" component if the network is still loading. That's a different state than "we couldn't fetch the data for some reason" and should be reflected accordingly.

Comment (pdp2121) [reply src/containers/Network/UpgradeStatus.tsx]: Thanks Mayukha, update the code and also the PR description.

File: src/containers/Network/UpgradeStatus.tsx [hunk 3]
```diff
@@ -111,37 +115,31 @@ export const UpgradeStatus = () => {
   return (
     <div className="network-page">
       <Streams validators={vList} updateValidators={updateValidators} />
-      {validatorCount && (
+      {
         // @ts-ignore - Work around for complex type assignment issues
         <Hexagons data={validations} list={vList} />
```
Comment (mvadari) [inline src/containers/Network/UpgradeStatus.tsx:120]: What's the type difficulty here?

Comment (ckniffen) [reply src/containers/Network/UpgradeStatus.tsx:120]: There is weirdness with `connect` where typescript interprets the component as having no props

**Pull request merged**